libraries/wasm-jit/Source/WAST/ParseNumbers.cpp

```diff
@@ -229,7 +229,7 @@ bool tryParseInt(ParseState& state,UnsignedInt& outUnsignedInt,I64 minSignedValu
 	{
 	case t_decimalInt:
 		isNegative = parseSign(nextChar);
-		u64 = parseDecimalUnsignedInt(nextChar,state,isNegative ? U64(-minSignedValue) : maxUnsignedValue,"int literal");
+		u64 = parseDecimalUnsignedInt(nextChar,state,isNegative ? -U64(minSignedValue) : maxUnsignedValue,"int literal");
```

Comment:
This is kind of interesting, as the old implementation `U64(-minSignedValue)` is exactly the UB case that we are trying to avoid. So I guess either:
- our compilers were doing what the new implementation does in that case
- or we never parsed values greater that the `maxValue` provided.

Reply:
> our compilers were doing what the new implementation does in that case

Probably, yeah. I think this is a case of "technically UB because it results in integer overflow.. but likely happens to do what we expect on platforms we care about". (we should still avoid it of course)

> or we never parsed values greater that the `maxValue` provided.

These two lines were hitting the following error so we were actually getting here,
> runtime error: negation of -9223372036854775808 cannot be represented in type 'I64' (aka 'long'); cast to an unsigned type to negate this value to itself